node/src/kdf.rs

```diff
@@ -41,3 +46,58 @@ pub fn derive_delta(receipt_id: CryptoHash, entropy: [u8; 32]) -> Scalar {
 pub fn derive_key(public_key: PublicKey, epsilon: Scalar) -> PublicKey {
     (<Secp256k1 as CurveArithmetic>::ProjectivePoint::GENERATOR * epsilon + public_key).to_affine()
 }
+
+#[derive(Debug)]
+pub struct MultichainSignature {
```

Comment:
should this be called MultichainEthSignature if only ETH is using the recovery id?